src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java

```diff
@@ -1548,20 +1549,51 @@ public void visitCtWildcardReference(CtWildcardReference wildcardReference) {
 
 	private boolean printQualified(CtTypeReference<?> ref) {
 		if (importsContext.isImported(ref)) {
-			// If my.pkg.Something is imported, but we are in the context of a class which is
-			// also called "Something", we should still use qualified version my.pkg.Something
+			// If my.pkg.Something is imported, but
+			//A) we are in the context of a class which is also called "Something",
+			//B) we are in the context of a class which defines field which is also called "Something",
+			//	we should still use qualified version my.pkg.Something
 			for (CtTypeReference<?> enclosingClassRef : context.currentThis) {
 				if (enclosingClassRef.getSimpleName().equals(ref.getSimpleName())
 						&& !Objects.equals(enclosingClassRef.getPackage(), ref.getPackage())) {
 					return true;
 				}
+				CtType<?> enclosingClass = enclosingClassRef.getDeclaration();
+				if (enclosingClass != null) {
+					CtField<?> field = getDeclareddOrInheritedField(enclosingClass, ref.getSimpleName());
+					if (field != null) {
+						return true;
+					}
+				}
 			}
 			return false;
 		} else {
 			return true;
 		}
 	}
 
+	private static CtField<?> getDeclareddOrInheritedField(CtType<?> type, String fieldName) {
```

Comment:
getDeclareddOrInheritedField -> getDeclaredOrInheritedField (double dd)

We agree to move it to CtType.
We must also handle fields from interfaces.